Include ROS distro information in server info message

bryfox opened the pull request:
### Changelog
Fix: ROS Foxglove bridge supports click-to-publish in 3d panel

### Docs

### Description

In order to support click-to-publish in the 3d panel on a WebSocket connection, the ServerInfo message needs to include the value of the ROS_DISTRO environment variable. (Publishing is only supported for ROS sources right now.) Prior versions of the Bridge included this but the new releases for ROS2 do not, and the SDK doesn't expose a mechanism for it.

This PR adds an undocumented `server_info` option to the Rust and C++ server options. This metadata will be included in the ServerInfo message sent to the app. The only use case today is to convey ROS distro, but we may extend in the future. I named this "server_info" instead of "metadata" since it's specifically for information sent to clients, but am open to better names.

This does not add python support, and we have no workarounds for people wanting to spoof support for 3d publishing there. ERT-689 covers this (bringing 3d publish support to non-ROS systems).

#### Alternatives considered

1. Set ROS_DISTRO in the SDK directly. This PR originally did this, which opens the door for easy workarounds for people, but has the downsides noted in comments.

2. Expose a ROS-specific option or some general metadata option, with no clear reason (or use) for the interface beyond the ROS distro. Since the server here is intended to serve the Foxglove app, including other arbitrary metadata isn't useful.

ericmlujan reviewed (commented):
+1 to @clalancette's opinion, I feel this lives more naturally in Foxglove Bridge. For some of our other work, I've also filed FLE-107 to expose `metadata` within ServerInfo from the SDK; Bridge could then use this.

bryfox commented:
It's unanimous, then. I updated the changes per suggestions above. TIL that we can use `@cond` on struct fields; I removed tls_identity from docs too since it was documented as internal — but we could leave that alone if desired.

I called the new option `server_info`; I think 'metadata' is too generic here, and want to convey somehow that this information will be sent to clients. Open to suggestions.

ericmlujan commented on cpp/foxglove/include/foxglove/server.hpp:
```diff
@@ -216,12 +216,20 @@ struct WebSocketServerOptions {
   FetchAssetHandler fetch_asset;
+  /// @brief A sink channel filter callback.
+  SinkChannelFilterFn sink_channel_filter;
+  /// @cond foxglove_internal
```
Did we want to make `tls_identity` foxglove internal? I thought this was shipped. It's in the documentation, at least?

bryfox commented on c/src/server.rs:
```diff
@@ -91,6 +94,14 @@ pub struct FoxgloveServerOptions<'a> {
+    /// - If provided, the `server_info` must be a valid pointer to an array of valid
+    ///   `FoxgloveKeyValue`s with `server_info_count` elements.
+    pub server_info: *const FoxgloveKeyValue,
+    pub server_info_count: usize,
```
I was following the example of `supported_encodings`/`supported_encodings_count` above (also in cloud sink). I think if we want to change these together in the future, we can do so freely.
